### DISCUSSION TITLE
[Feature Request] Variant Grouping

### DISCUSSION THREAD

**Discussion by nifte** [tailwindlabs/tailwindcss#7497] [Ideas]
It would be nice to be able to group classes that share the same variant(s) together without having to retype the varaint(s) for each individual class.

Example:
```html
<div class="md:dark:hover:(bg-blue-300,text-white,shadow-lg)" />
```
compared to what we have to do now:
```html
<div class="md:dark:hover:bg-blue-300 md:dark:hover:text-white md:dark:hover:shadow-lg" />
```

**Comment by adamwathan** [comment]
Hey! Something we've considered (and prototyped), but haven't decided if we should include it or not because the resulting CSS is bigger and less performant.

Check this Twitter thread for some details:

https://twitter.com/adamwathan/status/1461519820411789314

**Reply by nifte** [reply]
I see. Would you consider making it an opt-in config setting with a warning about the performance trade off? It would be really useful for small projects that don't have a ton of CSS but use several variants together.

And for larger projects, maybe let the CLI or VSCode plugin automatically expand the grouped variants to their individual classes? Not sure what the best solution is, but just an idea.
